Detectors/MUON/MID/Simulation/test/testSimulation.cxx

```diff
@@ -0,0 +1,208 @@
+o2::mid::DigitsMerger SIMUL::digitsMerger;
+o2::mid::PreClusterizer SIMUL::preClusterizer;
+o2::mid::Clusterizer SIMUL::clusterizer;
+
+std::vector<Hit> generateHits(size_t nHits, int deId, const o2::mid::Mapping& mapping, const o2::mid::GeometryTransformer geoTrans)
+{
+  std::vector<Hit> hits;
+  std::random_device rd;
+  std::mt19937 mt(rd());
+  std::uniform_real_distribution<double> distX(-127.5, 127.5);
+  std::uniform_real_distribution<double> distY(-40., 40.);
+  while (hits.size() < nHits) {
+    Point3D<float> point(distX(mt), distY(mt), 0.);
+    if (!mapping.stripByPosition(point.x(), point.y(), 0, deId, false).isValid()) {
+      continue;
+    }
+    Point3D<float> globalPoint = SIMUL::geoTrans.localToGlobal(deId, point);
+    hits.emplace_back(hits.size(), deId, globalPoint, globalPoint);
+  }
+  return hits;
+}
+
+bool checkLabel(const ColumnData& digit, MCLabel& label, std::string& errorMessage)
+{
+  std::stringstream ss;
+  ss << digit << "\n";
+  bool isOk = true;
+  if (label.getDEId() != digit.deId) {
+    isOk = false;
+    ss << "label deId " << label.getDEId() << " != " << digit.deId << "\n";
+  }
+  if (label.getColumnId() != digit.columnId) {
+    isOk = false;
+    ss << "label columnId " << label.getColumnId() << " != " << digit.columnId << "\n";
+  }
+  int firstStrip = label.getFirstStrip();
+  int lastStrip = label.getLastStrip();
+  int cathode = label.getCathode();
+  int nLines = (cathode == 0) ? 4 : 1;
+  for (int iline = 0; iline < nLines; ++iline) {
+    for (int istrip = 0; istrip < 16; ++istrip) {
+      int currStrip = MCLabel::getStrip(istrip, iline);
+      bool isFired = digit.isStripFired(istrip, cathode, iline);
+      if (isFired != (currStrip >= firstStrip && currStrip <= lastStrip)) {
+        isOk = false;
+        ss << "Cathode: " << cathode << "  firstStrip: " << firstStrip << "  lastStrip: " << lastStrip;
+        ss << "  line: " << iline << "  strip: " << istrip << "  fired: " << isFired << "\n";
+      }
+    }
+  }
+  errorMessage = ss.str();
+  return isOk;
+} // namespace mid
+
+std::vector<int> getDEList()
+{
+  std::vector<int> deList = { 2, 3, 4, 5, 6, 20, 21, 22, 23, 24, 47, 48, 49, 50, 51, 65, 66, 67, 68, 69 };
+  return deList;
+}
+
+BOOST_TEST_GLOBAL_FIXTURE(SIMUL);
+
+BOOST_AUTO_TEST_SUITE(o2_mid_simulation)
+// BOOST_FIXTURE_TEST_SUITE(sim, SIMUL)
+
+BOOST_DATA_TEST_CASE(MID_SimulChain, boost::unit_test::data::make(getDEList()), deId)
+{
+  // In this test, we generate a cluster from one impact point and we reconstruct it.
+  // If the impact point is in the RPC, the digitizer will return a list of strips,
+  // that are close to each other by construction.
+  // We will therfore have exactly one reconstructed cluster.
+  // It is worth noting, however, that the clustering algorithm is desigend to
+  // produce two clusters if the BP and NBP do not superpose.
+  // The digitizer produces superposed BP and NBP strips only if the response parameters
```

Comment:
looks like different tests to me (i.e. this test does too much)

also a few typos : therfore -> therefore ; desigend -> designed

(also, is the use of "superpose" instead of "overlap" intentional and/or actually worth noting ?)

Reply:
This is indeed a test of the chain. I wanted to be sure that if I start from 1 hit I get 1 recpoint.
Since the test is rather quick, it is quite useful to have it as a "unit_test". The alternative would be to fully remove this chain test.

Reply:
P.S. No specific meaning between superpose and overlap...

Reply:
> This is indeed a test of the chain. I wanted to be sure that if I start from 1 hit I get 1 recpoint.

Yet that's not the only case you test depending on params.getParB, right ? Alternatively I would argue that you should test for different combinations of getParB, and each alternative would be a test by itself.

Reply:
That's the easiest test I can do. When the parameters in the BP and NBP are the same, it means that the fire probability is the same, i.e. that I get a perfect overlap. In this situation, I only get 1 "rec point". On the other hand, if the parameters are different one might have no perfect overlap, and in this case I treat the non-overlapping parts as separate rec points. This is a feature.
A test of all cases is way beyond the scope of a "unit_test", while the test I'm performing, although not atomic, is a quite simple case that allows to spot possible issues.